src/components/core/geoChart/legends/PushpinCategoryLegend.tsx

```diff
@@ -1,26 +1,92 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
+import { ContentRect } from "react-measure";
+
+import FluidLegend from "../../../visualizations/chart/legend/FluidLegend";
 import StaticLegend from "../../../visualizations/chart/legend/StaticLegend";
+import { LEFT, RIGHT } from "../../../visualizations/chart/legend/PositionTypes";
+import { ChartType } from "../../../../constants/visualizationTypes";
 import { IPushpinCategoryLegendItem } from "../../../../interfaces/GeoChart";
 
+export const HEIGHT_OF_SIZE_LEGEND = 161;
+
 export interface IPushpinCategoryLegendProps {
-    categoryItems: IPushpinCategoryLegendItem[];
-    position: string;
-    onItemClick(item: IPushpinCategoryLegendItem): void;
+    categoryItems?: IPushpinCategoryLegendItem[];
+    chartType?: ChartType;
+    contentRect: ContentRect;
+    format?: string;
+    hasSizeLegend: boolean;
+    height?: number;
+    locale?: string;
+    position?: string;
+    responsive?: boolean;
+    showFluidLegend?: boolean;
+    onItemClick?(item: IPushpinCategoryLegendItem): void;
 }
 
-export default function PushpinCategoryLegend(props: IPushpinCategoryLegendProps): JSX.Element {
-    const { position, categoryItems, onItemClick } = props;
-    const classNames = `viz-static-legend-wrap s-geo-category-legend position-${position}`;
-    const DEFAULT_HEIGHT = 0;
-    const legendProps = {
-        series: [...categoryItems],
-        onItemClick,
-        position,
+export default class PushpinCategoryLegend extends React.PureComponent<IPushpinCategoryLegendProps> {
```

Comment:
It's still stateless component, right? So we should keep it as is?

Reply:
reverted :)